erpnext/schools/doctype/program_enrollment/program_enrollment.js

```diff
@@ -46,6 +46,7 @@ frappe.ui.form.on("Program Enrollment", {
 					}
 				}
 			});
+			frm.events.get_courses(frm);
```

Comment:
Both the functions is calling server side functions and executed asynchronously. In `get_courses` function, you are passing `doc` to the server, where still fees schedule is not there. Hence, if return of get_courses function comes later, it will overwrite the fees_schedule, which is earlier fetched from the first function.